src/toil/jobStores/fileJobStore.py

```diff
@@ -244,8 +247,12 @@ def jobs(self):
                 logger.warning('Job Dir: %s' % i)
                 if i.startswith(self.JOB_DIR_PREFIX):
                     # This is a job instance directory
+                    jobId = self._getJobIdFromDir(os.path.join(tempDir, i))
                     try:
-                        yield self.load(self._getJobIdFromDir(os.path.join(tempDir, i)))
+                        if self.exists(jobId):
+                            yield self.load(jobId)
+                        else:
+                            raise NoSuchFileException(jobId)
```

Comment:
This would be `NoSuchJobException` if you want to hit the `except` later. You could also just delete the whole `else` block.

As is it looks like when the job doesn't exist, you'll abort the whole workflow with a `NoSuchFileException`.